engine/execution/storehouse/executing_block_snapshot.go

```diff
@@ -58,6 +58,10 @@ func (s *ExecutingBlockSnapshot) getFromUpdates(id flow.RegisterID) (flow.Regist
 // Usually it's used to create a new storage snapshot at the next executed collection.
 // The registerUpdates contains the register updates at the executed collection.
 func (s *ExecutingBlockSnapshot) Extend(newCommit flow.StateCommitment, updates map[flow.RegisterID]flow.RegisterValue) execution.ExtendableStorageSnapshot {
+	if len(updates) == 0 {
+		return s
```

Comment:
whats the reasoning here? I'm not sure I understand.

Reply:
I will add comments, the idea is that if there is. no updates, it is not necessary to wrap with a snapshot that has no update, instead, just return the original snapshot itself, since it is able to return the same register.